sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import com.azure.android.communication.chat.implementation.notifications.NotificationUtils;
+import com.azure.android.communication.chat.models.ChatEvent;
+import com.azure.android.communication.chat.models.ChatEventType;
+import com.azure.android.communication.chat.models.ChatPushNotification;
+import com.azure.android.communication.chat.models.PushNotificationCallback;
+import com.azure.android.communication.common.CommunicationTokenCredential;
+import com.azure.android.core.logging.ClientLogger;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * The registrar client interface
+ */
+public class PushNotificationClient {
+    private final ClientLogger logger = new ClientLogger(PushNotificationClient.class);
+    private final CommunicationTokenCredential communicationTokenCredential;
+    private final RegistrarClient registrarClient;
+    private final Map<ChatEventType, Set<PushNotificationCallback>> pushNotificationListeners;
+    private boolean isPushNotificationsStarted;
+
+    public PushNotificationClient(CommunicationTokenCredential communicationTokenCredential) {
+        this.communicationTokenCredential = communicationTokenCredential;
+        this.pushNotificationListeners = new HashMap<>();
+        this.isPushNotificationsStarted = false;
+        this.registrarClient = new RegistrarClient();
+    }
+
+    /**
+     * flag to indicate if push notification has started
+     * @return boolean if push notification has started
+     */
+    public boolean hasStarted() {
+        return this.isPushNotificationsStarted;
+    }
+
+    /**
+     * Register the current device for receiving incoming push notifications via FCM.
+     * @param deviceRegistrationToken Device registration token obtained from the FCM SDK.
+     * @throws RuntimeException if push notifications failed to start.
+     */
+    public void startPushNotifications(String deviceRegistrationToken) {
+        if (this.isPushNotificationsStarted) {
+            return;
+        }
+
+        String skypeUserToken;
+        try {
+            skypeUserToken = communicationTokenCredential.getToken().get().getToken();
+        } catch (InterruptedException e) {
+            throw logger.logExceptionAsError(
+                new RuntimeException("Get skype user token failed for push notification: " + e.getMessage()));
+        } catch (ExecutionException e) {
+            throw logger.logExceptionAsError(
+                new RuntimeException("Get skype user token failed for push notification: " + e.getMessage()));
+        }
```

Comment:
We can collapse these catch blocks.
```suggestion
        } catch (ExecutionException | InterruptedException e) {
            throw logger.logExceptionAsError(
                new RuntimeException("Get skype user token failed for push notification: " + e.getMessage()));
        }
```